### ISSUE TITLE
Exiting from full-screen does not restore window after sleep/awake

### ISSUE THREAD

**Issue by HubKing** [mpv-player/mpv#11409]
Basically, the symptom is like the illustration below. I have an MPV window. I put the PC to sleep and then wake it up. I go into full-screen mode. Then I exit from the full-screen mode. I would expect the it would go back to the size/position before going into full-screen mode, but the window becomes big and goes out of the monitor boundary.

Is this a known problem? I am using Wayland Gnome 43.3 with two monitors that have different sizes/DPI's (`scale-monitor-framebuffer` is set).

![mpv](https://user-images.githubusercontent.com/10900489/222969299-ba0ce623-6f1e-4317-a00b-d83b4eb91247.png)

**Comment by Dudemanguy** [comment]
Post a log of this happening. I'm going to guess that on wakeup mpv gets bogus surface entrance events in which case there's probably not much we can do.

**Comment by HubKing** [comment]
> Post a log of this happening. I'm going to guess that on wakeup mpv gets bogus surface entrance events in which case there's probably not much we can do.

I attached the log of what this happened. For some reason, putting the computer to sleep did not work for a couple of times (monitor got turned off, but HDD lamp flashing) and sleep succeeded in the end, so probably the last part of the log contains that.

[output.txt](https://github.com/mpv-player/mpv/files/10892332/output.txt)

**Comment by Dudemanguy** [comment]
Something is very strange with this log. Why is it resizing so much so rapidly? Also, could you try git master? The code related to this is very different now, so I'm afraid trying to debug this won't help much.

**Comment by HubKing** [comment]
> Something is very strange with this log. Why is it resizing so much so rapidly? Also, could you try git master? The code related to this is very different now, so I'm afraid trying to debug this won't help much.

The MPV I had been using was the version in the Official repository: 1.0.35.1-2 (Build Date 21 Feb 2003). I uninstalled it and installed the [AUR mpv-git package](https://aur.archlinux.org/packages/mpv-git) (because I don't know how to manually build it) whose version is `0.35.0_135_g9b59d39a3a-1`. The problem does not seem to happen with the git version. But isn't the git version older (0.35.0 compared to 1.0.35.1-2)? The version numbers seems cryptic, so I am not sure which one is newer.

**Comment by HubKing** [comment]
I tested it again with the MPV that I built (I thought it would be complex and time-consuming, but it was rather easy and quick). The commands I used were

> git clone https://github.com/mpv-player/mpv.git
> meson setup build
> meson compile -C build
> sudo meson install -C build
> /usr/local/bin/mpv ~/Downloads/TextInMotion-VideoSample-1080p.mp4 --log-file=build3.txt

and still, the version is `mpv 0.35.0-306-g291e05ec1a` which seems older than repository's 0.35.1. When using `git clone` do I have to pass some special argument to get the latest?

Anyway, the problem occurs with this manual build. But it seems this depends on the window size *before* the sleep. I attach the log of when the problem happened.
[build3.txt](https://github.com/mpv-player/mpv/files/10892532/build3.txt)

**Comment by Dudemanguy** [comment]
No, the git version is newer. The git tags simply do not include the .1 number.

According to this log, fullscreen is never actually turned off. It's cycled on twice in a row. That doesn't seem right.
